server/src/main/java/org/opensearch/common/cache/stats/CacheStats.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.stats;
+
+import org.opensearch.core.common.io.stream.Writeable;
+
+import java.util.List;
+
+/**
+ * Interface for access to any cache stats. Allows accessing stats by dimension values.
+ */
+public interface CacheStats extends Writeable {// TODO: also extends ToXContentFragment (in API PR)
+
```

Comment:
To clarify, instances of `CacheStats` are the immutable snapshots of stats, right?

Not to be confused with `StatsHolder`, which holds the live counters?

Reply:
That's right. I'll clarify in the comments.